bin/import.ml

```diff
@@ -239,14 +239,16 @@ module Scheduler = struct
   ;;
 end
 
-let restore_cwd_and_execve (common : Common.t) prog argv env =
-  let prog =
-    if Filename.is_relative prog
-    then (
-      let root = Common.root common in
-      Filename.concat root.dir prog)
-    else prog
-  in
+let resolve_relative_path_within_root_dir (common : Common.t) path =
```

Comment:
you're threading `Common.t` all the way here just to get `Common.root`. how about we just take `~root` instead?

Reply:
fixed